src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/MenuItem.cs

```diff
@@ -946,7 +946,16 @@ private static void OnIsCheckedChanged(DependencyObject d, DependencyPropertyCha
         {
             MenuItem menuItem = (MenuItem) d;
 
-            if ((bool) e.NewValue)
+            bool oldValue = (bool)e.OldValue;
```

Comment:
Will the OldValue be `DependencyProperty.UnsetValue` ? And should we move the code to line 955?

Reply:
Yes, initially it is for the unset value and when we will toggle then it will take the previous value. And I moved to code as suggested.